### PR TITLE
Labyrinth reset tower

### PR DESCRIPTION
**Pull request by hypnoshock**
# What

Added a reset tower which will call the `reset` function on the `LabyrinthCore` contract. 

image

- This building is both present in the standalone and croissant maps
- A password must be supplied to reset
- The reset code has now been made a public function on the core building which can only be called by either the core building itself or the reset tower

# Why

Adds the ability to reset the Labyrinth without having to walk to the core building and supply 4 hearts

# Notes

If we want to change the password run a password (all lower case) through a keccak256 hash (can be done here: https://emn178.github.io/online-tools/keccak_256.html) and then update the `getPasswordHash` function in the `ResetTower` contract

```
    function getPasswordHash() internal pure returns (bytes32) {
        return 0x1f1fc8e191099524ede873ed4b43087e02baaa5f39243fdd808f60582dc7702e;
    }
```

### CODE REVIEW

File: contracts/src/maps/labyrinth/kinds/buildings/LabyrinthCore/LabyrinthCore.sol [hunk 1]
```diff
@@ -20,10 +21,21 @@ contract LabyrinthCore is BuildingKind {
     function use(Game ds, bytes24 buildingInstance, bytes24, /*actor*/ bytes memory /*payload*/ ) public override {
         // Crafting the Playtest Pass
         ds.getDispatcher().dispatch(abi.encodeCall(Actions.CRAFT, (buildingInstance)));
+        reset(ds, buildingInstance, buildingInstance);
```
Comment (chrisfarms) [inline contracts/src/maps/labyrinth/kinds/buildings/LabyrinthCore/LabyrinthCore.sol:24]: non-blocking: probably doesn't need to be doing a full CALL to itself, could be calling an internal func shared by both this and the public reset

File: contracts/src/maps/labyrinth/kinds/buildings/LabyrinthCore/ResetTower.sol [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import {Schema, Kind, Node} from "@ds/schema/Schema.sol";
+import {Actions} from "@ds/actions/Actions.sol";
+import {BuildingKind} from "@ds/ext/BuildingKind.sol";
+import {ILabyrinthCore} from "./ILabyrinthCore.sol";
+
+using Schema for State;
+
+contract ResetTower is BuildingKind {
+    function getPasswordHash() internal pure returns (bytes32) {
+        return 0x1f1fc8e191099524ede873ed4b43087e02baaa5f39243fdd808f60582dc7702e;
```
Comment (chrisfarms) [inline contracts/src/maps/labyrinth/kinds/buildings/LabyrinthCore/ResetTower.sol:16]: 🤫

File: contracts/src/maps/labyrinth/kinds/buildings/LabyrinthCore/ResetTower.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    );
+}
+
+function logState(state) {
+    console.log("State sent to pluging:", state);
+}
+
+const friendlyPlayerAddresses = [
+    // 0x402462EefC217bf2cf4E6814395E1b61EA4c43F7
+];
```
Comment (chrisfarms) [inline contracts/src/maps/labyrinth/kinds/buildings/LabyrinthCore/ResetTower.js:125]: **nit:** lots of these funcs seem unused, should we remove them?

File: contracts/src/maps/labyrinth/kinds/buildings/LabyrinthCore/ResetTower.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  inputs:
+    - name: Red Goo
+      quantity: 20
+    - name: Green Goo
+      quantity: 10
+    - name: Blue Goo
+      quantity: 2
+  outputs:
+    - name: Bronze Key
+      quantity: 1
```
Comment (chrisfarms) [inline contracts/src/maps/labyrinth/kinds/buildings/LabyrinthCore/ResetTower.yaml:25]: **nit:** these can be removed since it's `custom` right?

**Review by chrisfarms — APPROVED** [review]
set the password, and made it do the thing ... I'm probably not best placed to know if it did the _right_ thing, so maybe @5p0rt5BEArD can take a look ... but LGTM

couple of comments, but mostly just some tidy up of dead code

**Comment by hypnoshock** [comment]
Agreed, I'll fix these things before merge

File: contracts/src/maps/labyrinth/kinds/buildings/LabyrinthCore/LabyrinthCore.sol (hunk 1 above)
Comment (hypnoshock) [reply contracts/src/maps/labyrinth/kinds/buildings/LabyrinthCore/LabyrinthCore.sol]: Fixed. Now calls an internal `doReset` function. The public reset does the check to see if it's the reset building and if that passes calls the same internal `doReset`

File: contracts/src/maps/labyrinth/kinds/buildings/LabyrinthCore/ResetTower.yaml (hunk 4 above)
Comment (hypnoshock) [reply contracts/src/maps/labyrinth/kinds/buildings/LabyrinthCore/ResetTower.yaml]: Good point, all gone!

**Pull request merged**